docs/extensions/controllerregistration.md

```diff
@@ -226,7 +226,7 @@ The `globallyEnabled=true` option specifies that the `Extension/foo` object shal
 The `reconcileTimeout` tells Gardener how long it should wait during its shoot reconciliation flow for the `Extension/foo`'s reconciliation to finish.
 
 #### `Extension` Lifecycle
-The `lifecycle` field tells Gardener when to perform a certain action on the `Extension` resource during the reconciliation flows. If omitted, then the default behaviour will be applied. Please find more information on the defaults in the explanation below. Possible values for each control flow are `AfterKubeAPIServer` and `BeforeKubeAPIServer`. Let's take the following configuration and explain it.
+The `lifecycle` field tells Gardener when to perform a certain action on the `Extension` resource during the reconciliation flows. If omitted, then the default behaviour will be applied. Please find more information on the defaults in the explanation below. Possible values for each control flow are `AfterKubeAPIServer`, `BeforeKubeAPIServer`, and `AfterWorker`. Let's take the following configuration and explain it.
```

Comment:
I'm asking myself if we should have an e2e test similar to the existing test for extension lifecycles: https://github.com/gardener/gardener/blob/6c752067d0dcc13da715cd020fd1db7e86689a33/example/provider-local/garden/base/kustomization.yaml#L16-L31

Reply:
I am not sure, what this would actually test.

The manifest you linked is only for setting up the local provider, and not explicitly for testing the lifecycle strategies, right?

I also cannot find any existing E2E tests that are specifically looking at extensions and how / when they are reconciled. Writing a meaningful test, that actually asserts that the extension is reconciled at the right time also seems quite difficult, and a test that simple checks if the extension is reconciled at all adds little value IMO.

Reply:
The referenced extension types are used in the default e2e test shoot: https://github.com/gardener/gardener/blob/584ffb143fcd56d7cdb54ca53fd0da3b0ce807f9/test/e2e/gardener/common.go#L111
With this, the e2e tests assert that the extension can be successfully reconciled.

For the `AfterWorker` extension, we could deploy something to the shoot and wait for it to get ready (basically what this strategy is about). This way we would ensure that gardener fulfills and keeps fulfilling the promise of this feature.